File: src/main/java/dev/kensa/GivensBuilder.java
package dev.kensa;

import dev.kensa.state.CapturedInteractions;
import dev.kensa.state.Givens;

@FunctionalInterface
public interface GivensBuilder {
     default void build(Givens givens, CapturedInteractions interactions) {
          build(givens);
     }

     void build(Givens givens);
}

File: core/src/main/kotlin/dev/kensa/GivensBuilder.kt
package dev.kensa

import dev.kensa.state.Givens
import kotlin.DeprecationLevel.WARNING

@Deprecated("use Action<GivensBuilder> instead", ReplaceWith("Action<GivensContext>", "dev.kensa.Action", "dev.kensa.GivensContext"), WARNING)
@FunctionalInterface
fun interface GivensBuilder {
    fun build(givens: Givens)
}